client/file-manager/src/in_memory.rs

```diff
@@ -1,40 +1,167 @@
-use storage_hub_infra::types::{Chunk, FileProof, Key, Metadata};
+use std::collections::HashMap;
 
-use crate::traits::FileStorage;
+use common::types::HashT;
+use sp_core::H256;
+use storage_hub_infra::types::{Chunk, ChunkId, FileProof, Key, Leaf, Metadata};
 
-pub struct InMemoryFileStorage {}
+use sp_trie::{recorder::Recorder, MemoryDB, Trie, TrieDBBuilder, TrieLayout, TrieMut};
+use trie_db::TrieDBMutBuilder;
 
-impl InMemoryFileStorage {
+use crate::traits::{FileStorage, FileStorageError};
+
+pub struct FileData<T: TrieLayout + 'static> {
+    root: HashT<T>,
+    memdb: MemoryDB<T::Hash>,
+}
+
+impl<T: TrieLayout + 'static> FileData<T> {
+    pub fn get() {}
+}
+
+impl<T: TrieLayout + 'static> FileData<T> {
+    fn new() -> Self {
+        Self {
+            root: Default::default(),
+            memdb: MemoryDB::default(),
+        }
+    }
+
+    pub fn get_root(&self) -> H256 {
+        H256::from_slice(
```

Comment:
Why making `root` be something generic like `HashT<T>` but then making `get_root` return directly `H256`? In other words, why not just making `root: H256`, or going all the way with the generalisation?

Reply:
In the end we need to convert to the concrete type that the Runtime/other clients expect, which is `H256`. We have it as "generic" `HashT<T>` because that is what `TrieLayout` wants.